### ISSUE TITLE
Literal with computed union key is not type checked

### ISSUE THREAD

**Issue by kondi** [Microsoft/TypeScript#23206]
**TypeScript Version:**  2.9.0-dev.20180406

**Search Terms:** computed literal, computed property, string union key

**Code**

```ts
interface Obj {
  key1: string;
  key2: string; // required, there is no bug if Obj has only one possible key
}
type ObjKey = keyof Obj;
const key = 'key1' as ObjKey;  // type assertion is required, there is no bug if the type of key is the string literal

// No type error, 123 value is not checked 
const objLiteral: Partial<Obj> = {
  [key]: 123
};

// This is type checked correctly
const objAssign = {} as Partial<Obj>;
objAssign[key] = 123;
```

**Expected behavior:**

Type error for the `objLiteral` as well.

**Actual behavior:**

Only type error when assigning to `objAssign`, but `objLiteral` has no errors.

**Other information:**

I think the compiler thinks the `[key]` is just an excess property like if the key would be any other string, but it does not report the excess property because of #22427.

**Playground Link:** https://www.typescriptlang.org/play/#src=interface%20Obj%20%7B%0D%0A%20%20key1%3A%20string%3B%0D%0A%20%20key2%3A%20string%3B%20%2F%2F%20required%2C%20there%20is%20no%20bug%20if%20Obj%20has%20only%20one%20possible%20key%0D%0A%7D%0D%0Atype%20ObjKey%20%3D%20keyof%20Obj%3B%0D%0Aconst%20key%20%3D%20'key1'%20as%20ObjKey%3B%20%20%2F%2F%20type%20assertion%20is%20required%2C%20there%20is%20no%20bug%20if%20the%20type%20of%20key%20is%20the%20string%20literal%0D%0A%0D%0A%2F%2F%20No%20type%20error%2C%20123%20value%20is%20not%20checked%20%0D%0Aconst%20objLiteral%3A%20Partial%3CObj%3E%20%3D%20%7B%0D%0A%20%20%5Bkey%5D%3A%20123%0D%0A%7D%3B%0D%0A%0D%0A%2F%2F%20This%20is%20type%20checked%20correctly%0D%0Aconst%20objAssign%20%3D%20%7B%7D%20as%20Partial%3CObj%3E%3B%0D%0AobjAssign%5Bkey%5D%20%3D%20123%3B

**Related Issues:** #22427
**Not duplicate of:** #19938

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/13948

**Comment by kondi** [comment]
Oh, sorry then. I even checked quickly that issue before, but it looked different as he complains about getting an error when he should not, and in my case, this is the reverse. However, the reason looks the same, thanks for the reference.
